### DISCUSSION TITLE
Outdated copyright license in source code generation template?

### DISCUSSION THREAD

**Discussion by ThibFrgsGmz** [nasa/fprime#1306] [Q&A] [answered]
Hi,

I am extremely concerned about the [copyright](https://github.com/nasa/fprime/blob/devel/Autocoders/Python/src/fprime_ac/generators/templates/enums/enum_hpp.tmpl#L16) in the automatically generated source files.
The license model displayed is clearly copyleft and very contaminating/restrictive to the user's code, if not unusable to a non-US user:

```c
// Copyright (C) 2018 California Institute of Technology.
// ALL RIGHTS RESERVED.  United States Government Sponsorship
// acknowledged. Any commercial use must be negotiated with the Office
// of Technology Transfer at the California Institute of Technology.
//
// This software may be subject to U.S. export control laws and
// regulations.  By accepting this document, the user agrees to comply
// with all U.S. export laws and regulations.  User has the
// responsibility to obtain export licenses, or other export authority
// as may be required before exporting such information to foreign
// countries or providing access to foreign persons.
```

However, your framework is placed under the permissive [Apache 2.0 license](https://github.com/nasa/fprime/blob/devel/LICENSE.txt).

With this license, regardless of our country of origin, we can theoretically [[ref]](https://tldrlegal.com/license/apache-license-2.0-(apache-2.0)):
- make commercial use of the software.
- tinker with the software and make derivatives;
- distribute original or modified (derivative) works;
- use/modify the software freely without distributing it;
- grant/extend the license of the software;
- provide a warranty on the licensed software

In return, the user must only:

- incorporate the original copyright and the full text of the license into the modified software;
- report major changes to the software.

What is it really? Is the source file license a relic from a time when the framework was licensed under the Apache 2.0 license?

Could you tell us how the automatically generated source file license interacts with the Apache 2.0 license if it is still valid?

Because this license would be an undeniable obstacle to the use of the framework by my organization.

**Comment by LeStarch** [comment, marked as answer]
First, I agree that this is an extremely concerning issue, especially for corporations.  We'll get this sorted and I'll post back here. 

The license for F´  is Apache 2.0 and your notes on the license are consistent with my understanding of that license.  There are a few finer points to the Apache 2.0 license but those aren't relevant to this discussion.  The copyright of any files provided to you (as open source) should not constitute licensing.  If you see files in the repository containing out-of-date copyright, let us know and we will fix them.  These copyrights are such as:

https://github.com/nasa/fprime/blob/97f450a4d2705919a7b0f7d0bd98b06926192858/Svc/GenericHub/GenericHubComponentImpl.hpp#L6-L9

Here you can see the Caltech copyright header, and it has been stripped of any ITAR language.  This is because anything in the open source realm does not contain ITAR sensitive information. This is the copyright that the Apache 2.0 license requires you to maintain.

The copyright for the autocoded file outputs is clearly out-of-date and possibly erroneous. Here are the issues I need to look into:

1. Should a file generated by your team contain our team's Copyright?
2. ITAR language is not correct for the open source realm.

Also, if any similar issues come up that hinder the open source aspect of the software come up, let us know and we'll look into it!

**Reply by ThibFrgsGmz** [reply]
> Should a file generated by your team contain our team's Copyright?

From my user perspective, I would expect it to simply contain a reference to the Apache 2.0 license.
In source files, I personally prefer to keep the header to the bare minimum in order to have more direct access to the source code (i.e. not to have big comment tags).

Then, if the user wants to insert his own license, it seems to me that he could replace the copyright string in the autocoder according to his needs, no?

> Also, if any similar issues come up that hinder the open source aspect of the software come up, let us know and we'll look into it!

I have other concerns about this, but they are of a different nature. I hesitate to put them on the table because I feel like it's a Pandora's box, but if I find the time, I'll make a post on the forum.. These concerns are about governance, cybersecurity, and digital sovereignty, among others.

**Comment by LeStarch** [comment]
We agreed that these header in the autocoded output should be replaced with a header comment indicating that the file was autogenerated.  Projects can then list whatever copyright they need in something like a NOTICE.txt file.
